### PR TITLE
Fix Button Regression with null image source

### PR DESCRIPTION
**Pull request by tj-devel709**
This PR fixes an null reference issue that occurs when an button has its image set by UIButton.SetImage. Instead of using the Button.ImageSource.ToString() to see if the image has changed without needing to hold a reference to the image, the code now uses the UIButton.CurrentImage hash to determine if the image has changed. This helps us know if the image was changed in those situations where the Button.ImageSource is null but the UIButton.CurrentImage is still set.

### Issues Fixed

Fixes #25409

### CODE REVIEW

File: src/Controls/tests/TestCases.HostApp/Issues/Issue25409.xaml.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+	{
+		InitializeComponent();
+	}
+
+	private void OnCounterClicked(object sender, EventArgs e)
+	{
+#if IOS
+		((Action)(async () =>
+		{
+			var imageSource = new FileImageSource() { File = "shopping_cart.png" };
```
Comment (jsuarezruiz) [inline src/Controls/tests/TestCases.HostApp/Issues/Issue25409.xaml.cs:16]: Could the test be extended to:
- Set the image when the button is tapped.
- Remove the image, setting a null value when tapping again.
- Reset the image by tapping the button again (for example, using a counter, odd sets image, even removes it).
?

File: src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue25409.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
+using NUnit.Framework;
```
Comment (jsuarezruiz) [inline src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue25409.cs:1]: This specific test only have sense on iOS, right? Use compilation directives here.

**Review by jsuarezruiz — CHANGES_REQUESTED** [review]

**Comment by tj-devel709** [comment]
Okay, test has been rebased off the release/9.0.1xx branch, and it now uses the UIImage as a weakreference instead of the hash. Tests have also been updated on @jsuarezruiz's request!

File: src/Controls/src/Core/Button/Button.iOS.cs [hunk 3]
```diff
@@ -81,22 +80,29 @@ Size ICrossPlatformLayout.CrossPlatformMeasure(double widthConstraint, double he
 			}
 			platformButton.UpdateContentEdgeInsets(button, contentEdgeInsets);
 
+			_originalImageRef.TryGetTarget(out var _originalImage);
```
Comment (PureWeen) [inline src/Controls/src/Core/Button/Button.iOS.cs:83]: remove _ from this variable

**Review by PureWeen — CHANGES_REQUESTED** [review]
https://github.com/dotnet/maui/pull/25485/files#r1815631835

**Review by PureWeen — APPROVED** [review]

**Review by jsuarezruiz — APPROVED** [review]

**Pull request merged**